### DISCUSSION TITLE
grains boundaries overlap notindexed points

### DISCUSSION THREAD

**Discussion by niamaradi** [mtex-toolbox/mtex#1616] [Ask Anything]
Hello, 

I have a problem when computing grains. I think this problem is due to my Not indexed points. When I import my ebsd data, I assign points with a confidenceindex<0.2 to NotIndexed phase. Then I compute my grains on the indexed phase and I still get grain boundaries that overlap the not indexed points (see figure below). 

![image](https://user-images.githubusercontent.com/109603777/198536930-d7b9a5ad-980c-4de4-9782-25355e099f0e.png)

I also tried to assign the points that have a ci<0.2 to an empty dataset, the problem is still the same after computing grains. 

Also this is my command for computing grains

`[grains,ebsd.grainId] = calcGrains(ebsd('Ni'),'angle',[15 2]*degree,'boundary','tight');
`

Is there a way I can solve this problem ?
Thank you in advance for your help. 

Niama

**Comment by kilir** [comment]
Hi,
if you delete points from your dataset, they are not notIndexed but simply non-existent. If I understand you correctly, you want to restrict the positions of grain boundaries to be adjacent to indexed points. Instead of deleting them, you could for example set the points you do not like to the notIndexed phase and compute grains on the entire dataset.
See also here: https://mtex-toolbox.github.io/GrainReconstructionAdvanced.html

Cheers,
Rüdiger
